services/management/mongodb.go

```diff
@@ -44,10 +45,31 @@ func NewMongoDBService(db *reform.DB, registry agentsRegistry) *MongoDBService {
 func (s *MongoDBService) Add(ctx context.Context, req *managementpb.AddMongoDBRequest) (*managementpb.AddMongoDBResponse, error) {
 	res := new(managementpb.AddMongoDBResponse)
 
+	if err := validateNodeParamsOneOf(req.NodeId, req.NodeName, req.AddNode); err != nil {
+		return nil, err
+	}
+
 	if e := s.db.InTransaction(func(tx *reform.TX) error {
+		var nodeID string
+		switch {
+		case req.NodeId != "":
+			nodeID = req.NodeId
+		case req.NodeName != "":
+			node, err := models.FindNodeByName(tx.Querier, req.NodeName)
+			if err != nil {
+				return err
+			}
+			nodeID = node.NodeID
+		case req.AddNode != nil:
+			node, err := addNode(tx, req.AddNode, req.Address)
+			if err != nil {
+				return errors.Wrap(err, "can't create new node")
```

Comment:
If I understand that correctly, we should not wrap that error – it is already a good gRPC error we can return to the user.
API test should show that.

Reply:
Maybe we can extract that common code too.